ctl/proxy.go

```diff
@@ -0,0 +1,148 @@
+package ctl
+
+import (
+	"io"
+
+	kp "gopkg.in/alecthomas/kingpin.v2"
+)
+
+// Action is a CTL action
+type Action func() error
+
+// ParseContext is a proxy interface to kingpin's *ParseContext
+type ParseContext kp.ParseContext
+
+// CmdClause is a proxy interface to kingpin's *CmdClause
+type CmdClause kp.CmdClause
+
+// FlagClause is a proxy interface to kingpin's *FlagClause
+type FlagClause kp.FlagClause
+
+// Application is a proxy interface to kingpin's *Application
+type Application interface {
+	Version(version string) Application
+	Parse(args []string) (command string, err error)
+	Command(name, help string) *CmdClause
+	Flag(name, help string) *FlagClause
+	Terminate(terminate func(int)) Application
+	ErrorWriter(w io.Writer) Application
+	UsageWriter(w io.Writer) Application
+	//	ParseContext(args []string) (*ParseContext, error)
+}
+
+type proxyapp struct {
+	*kp.Application
+}
+
+// NewApplication creates a new application instance
+func NewApplication(name, help string) Application {
+	return &proxyapp{
+		Application: kp.New(name, help),
+	}
+}
+
+func (a *proxyapp) Terminate(terminate func(int)) Application {
+	a.Application.Terminate(terminate)
+	return a
+}
+
+func (a *proxyapp) Parse(args []string) (command string, err error) {
```

Comment:
How is that exported method does not have a comment?

Reply:
it's not `exported`. It's interface implementation on a private 
`(a *proxyapp)`